tests/shell/main.rs

```diff
@@ -0,0 +1,34 @@
+use lorri::{cas::ContentAddressable, ops::shell, project::Project, NixFile};
+use std::fs;
+use std::iter::FromIterator;
+use std::path::{Path, PathBuf};
+
+#[test]
+fn loads_env() {
+    let tempdir = tempfile::tempdir().expect("tempfile::tempdir() failed us!");
+    let project = project("loads_env", tempdir.path());
+    let output = shell::bash_cmd(project, tempdir.path())
+        .unwrap()
+        .args(&["-c", "echo $MY_ENV_VAR"])
+        .output()
+        .expect("failed to run shell");
+
+    assert_eq!(
+        // The string conversion means we get a nice assertion failure message in case stdout does
+        // not match what we expected.
+        String::from_utf8(output.stdout).expect("stdout not UTF-8 clean"),
+        "my_env_value\n"
+    );
+}
+
+fn project(name: &str, cache_dir: &Path) -> Project {
+    let test_root = PathBuf::from_iter(&[env!("CARGO_MANIFEST_DIR"), "tests", "shell", name]);
```

Comment:
Instead of that you can just use a `ContentAddressable` on a tempdir and reify an inline string from the test into a file. Then the files contents are directly here in the test and it’s easier to read.

Reply:
I'll give that a go, thanks.